docs/learn.md

```diff
@@ -721,20 +723,1250 @@ template](https://github.com/melange-re/melange-opam-template).
+external get_by_id : document -> string -> Dom.element = "getElementById"
+  [@@bs.send]
+
+let el = get_by_id document "my-id"
+```
+
+```js
+var el = document.getElementById("my-id");
+```
+
+When using `bs.send`, the first argument will be the object that holds the
+property with the function we want to call. In the example above, we are binding
+to the JavaScript object `document` using an abstract type, a very useful
+technique for bindings but also in other use cases. The "Encapsulation" section
+[in the OCaml Cornell
+textbook](https://cs3110.github.io/textbook/chapters/modules/encapsulation.html)
+provides a great overview to understand what abstract types are and when they
+are useful.
+
+##### Chaining
+
+It is common to find this kind of API in JavaScript: `foo().bar().baz()`. This
+kind of API can be designed with Melange externals and the `bs.send` attribute,
+in combination with the [the pipe operator](todo-fix-me.md).
+
+#### Variadic function arguments
+
+Sometimes JavaScript functions take an arbitrary amount of arguments. For this
+cases, Melange provides the `bs.variadic` attribute, which can be attached to
+the `external` declaration. However, there is one caveat: all the variadic
+arguments need to belong to the same type.
+
+```ocaml
+external join : string array -> string = "join"
+  [@@bs.module "path"] [@@bs.splice]
+let v = join [| "a"; "b" |]
+```
+
+Generates:
+
+```js
+var Path = require("path");
+var v = Path.join("a", "b");
+```
+
+#### Bing to a polymorphic function
+
+Some JavaScript libraries will define functions that where the arguments can
+vary on both type and shape.There are two approaches to bind to those, depending
+on how dynamic they are.
+
+##### Approach 1: Multiple external functions
+
+If it is possible to enumerate the many forms an overloaded JavaScript function
+can take, a flexible approach is to bind to each form individually:
+
+```ocaml
+external drawCat : unit -> unit = "draw" [@@bs.module "MyGame"]
+external drawDog : giveName:string -> unit = "draw" [@@bs.module "MyGame"]
+external draw : string -> useRandomAnimal:bool -> unit = "draw"
+  [@@bs.module "MyGame"]
+```
+
+Note how all three externals bind to the same JavaScript function, `draw`.
+
+##### Approach 2: Polymorphic variant + `bs.unwrap`
+
+In some cases, the function has a constant number of arguments but the type of
+the argument can vary. For cases like this, we can model the argument as a
+variant and use the `bs.unwrap` attribute in the external.
+
+Let’s say we want to bind to the following JavaScript function:
+
+```js
+function padLeft(value, padding) {
+  if (typeof padding === "number") {
+    return Array(padding + 1).join(" ") + value;
+  }
+  if (typeof padding === "string") {
+    return padding + value;
+  }
+  throw new Error(`Expected string or number, got '${padding}'.`);
+}
+```
+
+Here, `padding` can be modeled as a polymorphic variant.
+
+```ocaml
+external padLeft:
+  string
+  -> ([ `Str of string
```

Comment:
requirements:

- needs to be a polyvariant
- needs to contain payload

Reply:
Added in https://github.com/melange-re/documentation-site/pull/12/commits/11e3774e272a96fa359980d5fbaf4d76f7bd7a53, I also included the 2 other requisites.